declarative-labeler/action.yml

```
@@ -0,0 +1,122 @@
+name: 'Declarative Labeler'
+author: 'Brenno Lemos'
+description: 'Action para adicionar e remover labels de PRs e issues baseado em regras declarativas.'
+
+inputs:
+  github_token:
+    description: 'Token do GitHub. Geralmente estará em `secrets.GITHUB_TOKEN`. Necessário porque a API do GitHub necessita de permissões para alterar Pull Requests.'
+    required: true
+  
+  when_review_status:
+    description: 'Checa o status de aprovação do PR. O status de aprovação só é disparado caso o número de reviews configuradas no repositório seja atendida. Por padrão, é ignorado.'
+    choices:
+      - APPROVED
+      - REVIEW_REQUIRED
+      - CHANGES_REQUESTED
+      - ignore
+    default: ignore
+  
+  when_pr_is:
+    description: 'Checa se o PR é draft ou não. Os valores aceitos são `draft` e `ready`. O padrão é `ready`.'
+    choices:
+      - draft
+      - ready
+    default: ready
+  
+  then_add_labels:
+    description: 'Uma string separada por vírgulas (sem espaços entre elementos) com os nomes dos labels que serão adicionados. O padrão é vazio, ou seja, nenhum.'
+    default: ''
+  
+  then_remove_labels:
+    description: 'Uma string separada por vírgulas (sem espaços entre elementos) com os nomes dos labels que serão removidos. O padrão é vazio, ou seja, nenhum.'
+    default: ''
+  
+  also_reverse:
+    description: |
+      Define se a action também deve rodar em modo reverso. Se sim, a action inverterá os labels a serem adicinados e removidos no caso de alguma condição falhar.
+      Por exemplo, quando `true`, a action pode ser utilizada para adicionar o label `waiting QA` e remover o label `waiting review`, quando o PR é aprovado 2 vezes,
+      mas que também remove os label `waiting QA` e adiciona o label `waiting review` de volta quando uma ou mais das aprovações são *dismissed*. O padrão é `false`.
+
+    default: false
+
+runs:
+  using: composite
+  steps:
+
+    - name: Setup | Environment
+      shell: bash
+      run: |
+        echo "PASSING=true" >> "$GITHUB_ENV"
+        echo "TRIGGER_SIDE_EFFECTS=false" >> "$GITHUB_ENV"
+        echo "SIDE_EFFECT_ADD_LABELS=${{ inputs.then_add_labels }}" >> "$GITHUB_ENV"
+        echo "SIDE_EFFECT_REMOVE_LABELS=${{ inputs.then_remove_labels }}" >> "$GITHUB_ENV"
+
+    - name: Pre-When | Check if Review Decision is Approved
+      id: review-decision
+      if: env.PASSING == 'true' && startsWith(github.event_name, 'pull_request') && inputs.when_review_status != 'ignore'
```

```suggestion
      if: startsWith(github.event_name, 'pull_request') && inputs.when_review_status != 'ignore'
```
PASSING está sempre true aqui

---

De fato, mas de modo geral eu fiz com que as condições e consequências sejam independentes de si (tirando os nomeados `Pre-X`). Colocando a condição mesmo no primeiro step, eu garanto que a adição futura de novas condições funcione em qualquer ordem.

O mesmo pode ser dito da condição de baixo:

```yml
- name: When | Approved
  if: env.PASSING == 'true' && steps.review-decision.outputs.review_decision != inputs.when_review_status
```

Teoricamente não é necessário checar o `env.PASSING` já que a única maneira de ter um output em `steps.review-decision.outputs.review_decision` é que essa variável seja `'true'`. Contudo, dado que actions de GitHub não são testáveis, resolvi pecar pela redundância com esses checks de sanidade.